### DISCUSSION TITLE
How to set multiple 'Set-cookie' in route handler in app directory?

### DISCUSSION THREAD

**Discussion by DannyPark1** [vercel/next.js#48440] [Help]
### Summary

import { cookies } from 'next/headers'

export async function GET(request: Request) {
  const cookieStore = cookies();
  const token = cookieStore.get('token');

  return new Response('Hello, Next.js!', {
    status: 200,
    headers: { 'Set-Cookie': `token=${token}` }
  });
}

above the code is an example code for setting 'set-cookie' in route handler
appending multiple set-cookie results into one Set-cookie that has multiple key and value which is what i want to avoid.
How to set multiple 'Set-cookie' in route handler in app directory?

**Comment by philwolstenholme** [comment]
Didn't https://github.com/vercel/next.js/pull/47718 fix this? What version are you on?

**Reply by DannyPark1** [reply]
Im using version 13.2.4, and i'm looking for an example code where it can set multiple set-cookie with settings the cookies Path, samesite, domain attribute as well

**Reply by prashantchothani** [reply]
Even I am having the issue to set multiple cookies with Expires, Same Site Http only. Object iterator error is thrown. An example on Nextjs documentation for setting multiple cookies of these kind in Route Handler (app directory) would be useful. I am using 13.4.0

**Reply by ghost** [reply]
I used the below code to add multiple cookies.

```jsx

const newHeaders = new Headers(response.headers);

 // Setting accessToken as a cookie for easy access
newHeaders.set('set-cookie', `accessToken=${response.data.access_token};path=/;expires=${response.data.expires_in * 10};`);
// User id also saving in cookies
newHeaders.append('set-cookie', `userId=${response.data.userId};path=/;expires=${request.data.expires_in * 1000};`);

return NextResponse.json({ ...request.data }, { status: request.data.statusCode, headers: newHeaders });

```
